core-web/libs/sdk/analytics/src/lib/shared/analytics.utils.ts

```diff
@@ -0,0 +1,79 @@
+import { ANALYTICS_SOURCE_TYPE } from './analytics.constants';
+import { DotAnalyticsConfig, PageViewEvent } from './analytics.model';
+
+/**
+ * Retrieves analytics attributes from a given script element.
+ *
+ * @return {DotAnalyticsConfig | null} - The analytics attributes or null if there are no valid attributes present.
+ */
+export const getDataAnalyticsAttributes = (location: Location): DotAnalyticsConfig => {
+    const script = getAnalyticsScriptTag();
+
+    const attributes = {
+        server: script.getAttribute('data-analytics-server') || location.href,
+        debug: script.hasAttribute('data-analytics-debug'),
+        autoPageView: script.hasAttribute('data-analytics-auto-page-view'),
+        key: script.getAttribute('data-analytics-key') || ''
+    };
+
+    return attributes;
+};
+
+/**
+ * Retrieves the analytics script tag from the document.
+ *
+ * @returns {HTMLScriptElement} - The analytics script tag.
+ */
+export const getAnalyticsScriptTag = (): HTMLScriptElement => {
+    const scripts = document.querySelector('script[data-analytics-server]');
+
+    if (!scripts) {
+        throw new Error('Analytics script not found');
```

Comment:
Maybe we can do something like: 

`Dot Analytics: Script not found`

To maintain consistency with the other errors, a custom error can be helpful